pages/api/users/[uuid].ts

```diff
@@ -23,13 +27,27 @@ export default async function handlePrisma(
   res: NextApiResponse,
 ) {
   try {
+    const validationRequest = await validateRequest(req, res);
+    if (!validationRequest.session || !validationRequest.user) {
+      throw new HttpError('You are unauthorized!', 401);
+    }
     const reqHandler = req.method !== undefined && HANDLERS[req.method];
     if (reqHandler) {
-      if (typeof req.query.uuid !== 'string') {
-        throw new HttpError('Invalid ID', 400);
+      const queryUUID = z.string().safeParse(req.query.uuid);
+      if (!queryUUID.success) {
+        throw new HttpError(
+          'Invalid UUID! It should be given as a string!',
+          400,
+        );
```

Comment:
Tästä en oo ihan 100% varma miksi tein näin :smile: Tuohan näyttäisi olevan sama asia :smile:

Reply:
Lopputulos on sama. Ja halutessaan zodilla voi tehdä tarkempia tarkistuksia, esim. pituuden on vastattava uuid:n pituutta.

Reply:
Hyvä idea! Tsekkasin netistä ja tietokannasta, että UUID:n pituus on 36 merkkiä (32 merkkiä ja 4 viivaa). Lisäsin  Zodin tsekkaukseen tarkaksi pituudeksi (`.length`-methodilla) 36 merkkiä :+1:

Reply:
Katsos vaan, tuo linkki olikin hyvä. Jostain syystä mennyt ohi kokonaan :smile:  Kiitos! :+1: